### ISSUE TITLE
Feature request: max-lineheight option

### ISSUE THREAD

**Issue by majuscule** [peco/peco#471]
# What is the peco version, OS, architecture? 

N/A

# Abstract

Feature request - it would be nice to have a max-lineheight option for cases where you want to see some of your immediate history while reverse searching shell history. For example, if I have just typed a command with a kubernetes pod name (which includes a hash identifier) and I'd like to be able to reference when writing my search query.

**Comment by lestrrat** [comment]
I'm not sure I get it.

So suppose you had a search buffer like

```
a
b
c
d
e
... (ad infinitum)
```

and you were positioned at "c", you can, for example, see

```
b
c
d
```

?

**Comment by majuscule** [comment]
Not quite. So at the moment, when you run `peco file`, you get a prompt at the bottom of the screen, and the rest of the terminal is used for showing lines. I would like to restrict the portion of the terminal that peco takes over. i.e.:

right before executing peco:
```
$ foo # 1 
$ bar # 2
$ baz # 3
$ bat # 4
$ bla # 5
$ alb # 6
$ oof # 7
$ peco --max-lineheight 5 ~/.history
```

current behavior (note that the entire terminal is used by peco):
```
foo # 1 
bar # 2 
baz # 3
bat # 4
bla # 5
alb # 6
oof # 7
QUERY>
```

desired peco behavior (note that peco is using only 5 lines of the terminal to scroll through input):
```
$ foo # 1
$ bar # 2
baz # 3
bat # 4
bla # 5
alb # 6
oof # 7
QUERY>
```

Ideally, this also implies that peco would execute from it's current location in the terminal, rather than jumping to the bottom. i.e.:

```
$ foo # 1 
$ bar # 2
$ baz # 3
$ bat # 4
$ bla # 5
$ alb # 6
$ oof # 7
$ peco --max-lineheight 5 ~/.history








```

Would result in:
```
$ foo # 1 
$ bar # 2 
baz # 3
bat # 4
bla # 5
alb # 6
oof # 7
QUERY>








```

instead of the current behaviour:
```









foo # 1 
bar # 2 
baz # 3
bat # 4
bla # 5
alb # 6
oof # 7
QUERY>
```

Hopefully that all makes sense! I realize that this might be rather difficult to implement technically :-). Thanks for your work on the project!

**Comment by lestrrat** [comment]
I get it. Unfortunately our current underlying terminal control library (termbox-go) seems to "clear" the screen upon being initialized, so I think it's a no go ATM. However, I have seen in the past that termbox-go's development has been semi-suspended and that it might be time to move to a new library. Maaaaaaaaaaybe a new library would give us this feature of only hijacking part of the terminal (/cc @mattn )

**Comment by mattn** [comment]
@majuscule You probably have interesting in https://github.com/mattn/cho .

**Comment by tomykaira** [comment]
+1 to this issue.

I found the same request not from functionality point of view, but from performance.
When the terminal window height is large (more than 100 lines) and terminal rendering is slow, for example over SSH or WSL, incremental search experience is bad.
I should wait until all data are loaded and screen is refreshed.
If we have max line height option, in other words, if peco uses only 20 lines for example, I believe rendering performance becomes adequate.

I am happy to try to implement if contributors think it worth.
